pkg/controller/cluster/clusterapi_cluster_controller.go

```diff
@@ -0,0 +1,335 @@
+	// clustersSynced returns true if the cluster shared informer has been synced at least once.
+	// Added as a member to the struct to allow injection for testing.
+	clustersSynced cache.InformerSynced
+
+	// machineSetsLister is able to list/get machine sets and is populated by the shared informer passed to
+	// NewController.
+	machineSetsLister listers.MachineSetLister
+	// machineSetsSynced returns true if the machine set shared informer has been synced at least once.
+	// Added as a member to the struct to allow injection for testing.
+	machineSetsSynced cache.InformerSynced
+
+	// Clusters that need to be synced
+	queue workqueue.RateLimitingInterface
+
+	logger log.FieldLogger
+}
+
+func (c *CAPIController) addCluster(obj interface{}) {
+	cluster := obj.(*clusterapi.Cluster)
+	colog.WithClusterAPICluster(c.logger, cluster).Debugf("adding cluster")
+	c.enqueueCluster(cluster)
+}
+
+func (c *CAPIController) updateCluster(old, cur interface{}) {
+	oldCluster := old.(*clusterapi.Cluster)
+	curCluster := cur.(*clusterapi.Cluster)
+	colog.WithClusterAPICluster(c.logger, oldCluster).Debugf("updating cluster")
+	c.enqueueCluster(curCluster)
+}
+
+func (c *CAPIController) deleteCluster(obj interface{}) {
+	cluster, ok := obj.(*clusterapi.Cluster)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+			return
+		}
+		cluster, ok = tombstone.Obj.(*clusterapi.Cluster)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a Cluster %#v", obj))
+			return
+		}
+	}
+	colog.WithClusterAPICluster(c.logger, cluster).Debugf("deleting cluster")
+	c.enqueueCluster(cluster)
+}
+
+// When a machine set is created, enqueue the cluster that
+func (c *CAPIController) addMachineSet(obj interface{}) {
+	machineSet := obj.(*clusterapi.MachineSet)
+
+	if machineSet.DeletionTimestamp != nil {
+		return
+	}
+
+	cluster, err := controller.ClusterForClusterAPIMachineSet(machineSet, c.clustersLister)
+	if err != nil {
+		glog.V(2).Infof("error retrieving cluster for machine set %q/%q: %v", machineSet.Namespace, machineSet.Name, err)
+		return
+	}
+	if cluster == nil {
+		// If the cluster owner reference is not set, attempt to retrieve the cluster by label
+		if machineSet.Labels != nil {
+			if clusterName, ok := machineSet.Labels[clusterAPIClusterNameLabel]; ok {
+				cluster, err = c.clustersLister.Clusters(machineSet.Namespace).Get(clusterName)
+				if err != nil {
+					glog.V(2).Infof("error retrieving cluster for machineset %s/%s: %v", machineSet.Namespace, machineSet.Name, err)
+					return
+				}
+			}
+		}
+		if cluster == nil {
+			return
+		}
+	}
+	colog.WithClusterAPIMachineSet(colog.WithClusterAPICluster(c.logger, cluster), machineSet).Debugln("machineset created")
+	c.enqueueCluster(cluster)
+}
+
+// When a machine set is updated, figure out what cluster manages it and wake it
+// up.
```

Comment:
Is it ok for these comments to ignore the normal godoc standard of method name first because they're not exported? (for my own knowledge most likely)

Reply:
I assume it is ok because none of our linting tools catch it as a problem.